### PR TITLE
 python3Packages.secretstorage: 2.3.1 -> 3.1.0

### PR DESCRIPTION
**Pull request by worldofpeace**
#### Changelog: https://github.com/mitya57/secretstorage/blob/master/changelog

Pertinent changes:

- ported from dbus-python to jeepney
  Thusly Python 3.5 or newer is required.
- We can use pypi
- Add pygobject3

###### Motivation for this change
This comment from @Mic92 https://github.com/NixOS/nixpkgs/pull/51269#issuecomment-445198589

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/python-modules/jeepney/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+, pytest
+, testpath
+, tornado_4
+}:
+
+buildPythonPackage rec {
+  pname = "jeepney";
+  version = "0.4";
+
+  disabled = !(pythonAtLeast "3.5");
```
Comment (Mic92) [inline pkgs/development/python-modules/jeepney/default.nix]:
```suggestion
  disabled = pythonOlder "3.5";
```

File: pkgs/development/python-modules/secretstorage/default.nix [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
   pname = "secretstorage";
-  version = "2.3.1";
+  version = "3.1.0";
 
-  src = fetchFromGitHub {
-    owner = "mitya57";
-    repo = "secretstorage";
-    rev = version;
-    sha256 = "1sjd2jjbxgkkxyrfwx89x0hsnn39w2cr2qkxbg1iz52znr4sqism";
+  disabled = !(pythonAtLeast "3.5");
```
Comment (Mic92) [inline pkgs/development/python-modules/secretstorage/default.nix]:
```suggestion
  disabled = pythonOlder "3.5";
```

**Comment by worldofpeace** [comment]
@GrahamcOfBorg build python3Packages.secretstorage python3Packages.keyring

File: pkgs/development/python-modules/jeepney/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0w1w1rawl9k4lx91w16d19kbmf1349mhy8ph8x3w0qp1blm432b0";
+  };
+
+  checkInputs = [
+    pytest
+    testpath
+    tornado_4
```
Comment (dotlambda) [inline pkgs/development/python-modules/jeepney/default.nix]: Where is this requirement specified?
It's really sad to see people adding different versions of python packages to `python-packages.nix` again and again. (I know it wasn't you).

Comment (worldofpeace) [reply pkgs/development/python-modules/jeepney/default.nix]: > Where is this requirement specified?

My mistake, this should be `tornado` and a `propagatedBuildInputs`.
Seems to be a sort of integration https://gitlab.com/takluyver/jeepney/blob/master/docs/integrate.rst

> It's really sad to see people adding different versions of python packages to python-packages.nix again and again. (I know it wasn't you).

With enough frequency we should probably add a check to OfBorg for it.

**Comment by worldofpeace** [comment]
@dotlambda I've corrected https://github.com/NixOS/nixpkgs/pull/51904#discussion_r241008381
Also noticed that jeepney didn't have a license so i added that.

So here's the things I've done to test:
* Ran `nix-review` successfully (unrelated cached failures ignored)
* Tested examples for jeepney in a running session with dbus

And did what I was here to do in the first place, which was to make `pgcli` execute successfully :+1:

**Pull request merged**